tests/test_keyio.py

```diff
@@ -430,3 +431,44 @@ def test_get_signing_key_use_undefined():
 
     keys = kj.get_signing_key(key_type='rsa', kid='rsa1')
     assert len(keys) == 1
+
+
+KEYDEFS = [
+    {"type": "RSA", "key": '', "use": ["sig"]},
+    {"type": "EC", "crv": "P-256", "use": ["sig"]}
+]
+
+
+def test_remove_after():
+    # initial keyjar
+    keyjar = build_keyjar(KEYDEFS)[1]
+    _old = [k.kid for k in keyjar.get_issuer_keys('') if k.kid]
+    assert len(_old) == 2
+
+    # rotate_keys = create new keys + make the old as inactive
+    keyjar = build_keyjar(KEYDEFS, keyjar=keyjar)[1]
+
+    keyjar.remove_after = 1
+    # None are remove since none are marked as inactive yet
+    keyjar.remove_outdated()
+
+    _interm = [k.kid for k in keyjar.get_issuer_keys('') if k.kid]
+    assert len(_interm) == 4
+
+    # Now mark the keys to be inactivated
+    _now = time.time()
+    for k in keyjar.get_issuer_keys(''):
+        if k.kid in _old:
+            if not k.inactive_since:
+                k.inactive_since = _now
+
+    time.sleep(2)
```

Comment:
Oh! This is generally a bad idea. The slower the tests get, the more likely we are not to run them. I would absolutely recommend using [freezegun](https://github.com/spulec/freezegun) and a [manual tick](https://github.com/spulec/freezegun#manual-ticks) (which bumps time along instantly!).